### ISSUE TITLE
False positive on gethostname in socket module

### ISSUE THREAD

**Issue by dalemyers** [PyCQA/pylint#4302]
### Steps to reproduce

Create a simple file (or use this zipped example [socket_test.zip](https://github.com/PyCQA/pylint/files/6264705/socket_test.zip)):

```python
"""Test module."""

import socket

print(socket.gethostname())

```

Install pylint (you can use the following requirements.txt file to get the latest, or install the pre-release version):

```
astroid==2.5.2
isort==5.8.0
lazy-object-proxy==1.6.0
mccabe==0.6.1
pylint==2.7.4
toml==0.10.2
wrapt==1.12.1
```

Run `pylint test.py` on it. 

### Current behavior

```
************* Module test
test.py:5:6: E1101: Module 'socket' has no 'gethostname' member (no-member)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -15.00/10, +0.00)

```

### Expected behavior
No issues will be found

### pylint --version output

Result of `pylint --version` output:
```
pylint 3.0.0a1
astroid 2.5.2
Python 3.9.2 (default, Mar 26 2021, 23:22:38)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

or

```
pylint 2.7.4
astroid 2.5.2
Python 3.9.2 (default, Mar 26 2021, 23:22:38)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

Both repro. 

Note: Running this on macOS Big Sur

**Comment by hippo91** [comment]
@dalemyers thanks for your report. I do not understand clearly what is the problem here as the `Current behavior` section is empty.
However when linting the snippet you gave, `pylint`doesn't complain about anything.

**Comment by dalemyers** [comment]
Updated the current behavior section (I'd accidentally pasted my output in the HTML comment in the template). 

Everyone on my team is seeing this same behavior. 

Here are the _exact_ steps I took to reproduce on a brand new machine (only took it out of the box yesterday):

1. Create a new folder somewhere called whatever you like
2. Create a new python file inside that called "test.py" (this can really be whatever you like as well)
3. Set the content of that to the example shown above
3. Create a new virtual environment `python3 -m venv venv`
4. Activate the virtual environment `source venv/bin/activate`
5. Install pylint `python3 -m pip install pylint`
6. Run pylint `python3 -m pylint test.py`

Python 3.9.4 in this case (repro above was 3.9.2).

**Comment by ravisumit33** [comment]
I also can reproduce this issue. 
@dalemyers I am facing similar issue #4592 . Can you also reproduce it?

Is it common on Mac OS ?
